### PR TITLE
Sys 4333 new gateway endpoints

### PR DESCRIPTION
**Pull request by ivan-cholakov**

### CODE REVIEW

File: lib/apis/proxy.ts [hunk 1]
```diff
@@ -377,6 +381,93 @@ async function signProxySubmitCheckpointWithIdentity({relayer, signerAddress, ch
...
+
+    const encodedDataToSign = encodeOrderedData(orderedData);
+    return await signData(api, signerAddress, encodedDataToSign);
+  }
+
+async function signProxyBuy({ relayer, nonce, signerAddress, marketId, assetCount, asset, amountIn, maxPrice, orders, strategy, api }){
+  const dataRelayer = AccountUtils.convertToPublicKeyIfNeeded(relayer);
+
+  const orderedData = [
+    { Text: 'buy outcome tokens' },
```
Comment (nahuseyoum) [inline lib/apis/proxy.ts:423]: I thought we changed these to underscores

File: lib/apis/send.ts [hunk 2]
```diff
@@ -258,12 +258,89 @@ export class Send {
...
+  }
+// TODO: stub
+  async submitReport(handler: string, checkpoint: string, chainId: number): Promise<string> {
+    Utils.validateAccount(handler);
+    Utils.validateCheckpointFormat(checkpoint);
+    const methodArgs = { handler, checkpoint, chainId };
+    return await this.proxyRequest(methodArgs, TxType.ProxyReport, NonceType.PredictionMarkets);
+  }
+// TODO: stub
+  async submitCreateMarketAndDeployPool(handler: string, checkpoint: string, chainId: number): Promise<string> {
```
Comment (nahuseyoum) [inline lib/apis/send.ts:289]: Is this correct?

File: lib/apis/send.ts [hunk 3]
```diff
@@ -258,12 +258,89 @@ export class Send {
...
 
-  async submitCheckpoint(handler: string, checkpoint: string, chainId:number): Promise<string> {
+  async submitCheckpoint(handler: string, checkpoint: string, chainId: number): Promise<string> {
     Utils.validateAccount(handler);
     Utils.validateCheckpointFormat(checkpoint);
     const methodArgs = { handler, checkpoint, chainId };
     return await this.proxyRequest(methodArgs, TxType.ProxySubmitCheckpoint, NonceType.Anchor);
   }
+  // TODO: stub
+  async submitBuy(handler: string, checkpoint: string, chainId: number): Promise<string> {
```
Comment (nahuseyoum) [inline lib/apis/send.ts:268]: Are these placeholders?

File: lib/apis/send.ts [hunk 4]
```diff
@@ -381,13 +458,17 @@ export class Send {
...
+  private async getProxyNonce(nonceType: NonceType, requestId: string, proxyNonceData?: NonceData, nftId?: string, chainId?: number, marketId?: string) {
+    if (nonceType !== NonceType.Nft && nonceType !== NonceType.Anchor && nonceType !== NonceType.PredictionMarkets && nonceType !== NonceType.HybridRouter && !proxyNonceData) return undefined;
 
     if (nonceType === NonceType.Nft) {
       return new BN(await this.queryApi.getNftNonce(nftId)).toNumber();
-    } else if(nonceType === NonceType.Anchor) {
+    } else if (nonceType === NonceType.Anchor) {
       return new BN(await this.queryApi.getAnchorNonce(chainId)).toNumber();
+    } else if (nonceType === NonceType.PredictionMarkets) {
+      return new BN(await this.queryApi.getPredictionMarketsNonce(marketId, this.signerAddress)).toNumber();
```
Comment (nahuseyoum) [inline lib/apis/send.ts:469]: Don't we have 2 nonces? Which one is returned here?

File: lib/apis/send.ts [hunk 5]
```diff
@@ -258,12 +258,89 @@ export class Send {
...
+// TODO: stub
+  async submitCreateMarketAndDeployPool(handler: string, checkpoint: string, chainId: number): Promise<string> {
+    Utils.validateAccount(handler);
+    Utils.validateCheckpointFormat(checkpoint);
+    const methodArgs = { handler, checkpoint, chainId };
+    return await this.proxyRequest(methodArgs, TxType.ProxyCreateMarketAndDeployPool, NonceType.PredictionMarkets);
+  }
+
+  async createMarketAndDeployPool(baseAsset,
+    creatorFee,
```
Comment (nahuseyoum) [inline lib/apis/send.ts:297]: This should be hardcoded to 0

File: lib/apis/send.ts [hunk 6]
```diff
@@ -258,12 +258,89 @@ export class Send {
...
+    oracle,
+    period,
+    deadlines,
+    metadata,
+    marketType,
```
Comment (nahuseyoum) [inline lib/apis/send.ts:302]: This should be hardcoded to categorical

File: lib/apis/send.ts [hunk 7]
```diff
@@ -258,12 +258,89 @@ export class Send {
...
+    disputeMechanism,
```
Comment (nahuseyoum) [inline lib/apis/send.ts:303]: This should be hardcoded to Authorised

**Review by nahuseyoum — COMMENTED** [review]
The hardcoded market creation values should be removed from the api and instead be hardcoded on the gateway lambdas.

**Review by micaelffrancoAV — APPROVED** [review]

**Pull request merged**